specs/core/0_beacon-chain.md

```diff
@@ -1766,6 +1790,27 @@ For each `exit` in `block.body.exits`:
 * Verify that `bls_verify(pubkey=validator.pubkey, message_hash=exit_message, signature=exit.signature, domain=get_domain(state.fork, exit.epoch, DOMAIN_EXIT))`.
 * Run `initiate_validator_exit(state, exit.validator_index)`.
 
+##### Transfers
+
+Note: Transfers are a temporary functionality for phases 0 and 1, to be removed in phase 2.
+
+Verify that `len(block.body.transfers) <= MAX_TRANSFERS`.
+
+For each `transfer` in `block.body.transfers`:
+
+* Verify that `state.validator_balances[transfer.from] >= transfer.amount`.
+* Verify that `state.validator_balances[transfer.from] >= transfer.fee`.
+* Verify that `state.validator_balances[transfer.from] == transfer.amount + transfer.fee` or `state.validator_balances[transfer.from] >= transfer.amount + transfer.fee + MIN_DEPOSIT_AMOUNT`.
+* Verify that `transfer.slot == state.slot`.
+* Verify that `transfer.from` does not match the `from` field of another transfer in `block.body.transfers`.
+* Verify that `get_current_epoch(state) <= state.validator_registry[transfer.from].withdrawal_epoch`.
```

Comment:
In phase 0, these two conditions are always true.

We need to add a condition to ensure that the validator is exited.

```suggestion
* Verify that `get_current_epoch(state) >= state.validator_registry[transfer.to].exit_epoch`.
* Verify that `get_current_epoch(state) <= state.validator_registry[transfer.from].withdrawal_epoch`.
```

Reply:
`exit_epoch` doesn't work because we want to allow some time for validators to be slashed. My suggestion would be `exit_epoch + 2**13` which adds a security margin of 34 days.

Reply:
Ah, you're right. Need a new constant for this then. `MIN_EXIT_EPOCHS_BEFORE_TRANSFER`

Reply:
Note that the `MIN_EXIT_EPOCHS_BEFORE_TRANSFER` restriction is mainly for the sender (as opposed to the recipient).

Question: why restrict the recipient at all?

Reply:
Now that I think about it, you're right, the recipient doesn't need to be restricted...

I think what we're looking for is that the WITHDRAWABLE flag of the sender needs to be set.

Reply:
I've removed the constraint on the recipient. I've also added  `MIN_EXIT_EPOCHS_BEFORE_TRANSFER` as per above.

Reply:
> I think what we're looking for is that the WITHDRAWABLE flag of the sender needs to be set.

I'm tempted to remove the logic related to `WITHDRAWABLE` until phase 2. This includes removing [`process_withdrawals`](https://github.com/ethereum/eth2.0-specs/blob/d4f79c8878375a9e146f3491e2b4d3043a35a30a/specs/core/0_beacon-chain.md), `MIN_VALIDATOR_WITHDRAWAL_EPOCHS` and the `WITHDRAWABLE` flag.

Reply:
So we _do_ need a flag to say "we're done passing through the exit queue, we're now in the stage where we're waiting for custody challenges". And we _do_ also need a concept of "withdrawableness" because transferring ETH from one account to another _is_ a kind of withdrawal. 

My inclination is to take the withdrawable logic from the phase 1 PR and import it into phase 0. That is, when a validator exits the queue, their `withdrawable_epoch` gets set to `now + `MIN_VALIDATOR_WITHDRAWAL_EPOCHS`. This gets us handling for two state changes for the price of one.

Reply:
> My inclination is to take the withdrawable logic from the phase 1 PR and import it into phase 0.

Sounds good 👍 

Nomenclature side note: Talking about the "exit queue" is much better than the language we had before. Did some renaming in #534:

* `process_withdrawals` => `process_exit_queue`
* `MAX_WITHDRAWALS_PER_EPOCH` => `MAX_EXIT_DEQUEUES_PER_EPOCH`
* `withdrawn_so_far` => `dequeues`